lib/telemetry.go

```diff
@@ -399,9 +337,25 @@ func circonusSink(cfg TelemetryConfig, hostname string) (metrics.MetricSink, err
 	return sink, nil
 }
 
+// PrometheusDefs wraps collections of metric definitions to pass into the PrometheusSink
+type PrometheusDefs struct {
+	Gauges    []prometheus.GaugeDefinition
+	Counters  []prometheus.CounterDefinition
+	Summaries []prometheus.SummaryDefinition
+}
+
+// EmptyPrometheusDefs returns a PrometheusDefs struct where each of the slices have zero elements, but not nil.
+func EmptyPrometheusDefs() PrometheusDefs {
+	return PrometheusDefs{
+		Gauges:    []prometheus.GaugeDefinition{},
+		Counters:  []prometheus.CounterDefinition{},
+		Summaries: []prometheus.SummaryDefinition{},
+	}
+}
+
 // InitTelemetry configures go-metrics based on map of telemetry config
 // values as returned by Runtimecfg.Config().
-func InitTelemetry(cfg TelemetryConfig) (*metrics.InmemSink, error) {
+func InitTelemetry(cfg TelemetryConfig, defs PrometheusDefs) (*metrics.InmemSink, error) {
```

Comment:
Generally if a function accepts a config or options struct all the config would be part of that struct. This convention seems appropriate here as well.

Instead of a new `PrometheusDefs` struct, `TelemetryConfig` can use a `PrometheusOpts` field with type `prometheus.PrometheusOpts`. This would remove the need for the `PrometheusDefs` which closely mirrors an existing struct, and  would also also use to remove the `PrometheusRetentionTime` field, since that field is already in `PrometheusOpts`.

Reply:
Yeah I agree w/ that - the prometheus defs were a workaround to avoid adding an extra arg for each of the definition slices. I was a bit concerned about RuntimeConfig not mirroring the config we get in from disk like it does in many places. But adding the PrometheusOpts on there and translating it in `builder.go` does cut out of a bunch of boilerplate here.